app/vmauth/main.go

```diff
@@ -200,10 +200,11 @@ func processRequest(w http.ResponseWriter, r *http.Request, ui *UserInfo) {
 		up, hc = ui.DefaultURL, ui.HeadersConf
 		isDefault = true
 	}
+	r.Body = getReadTrackingBody(r.Body, int(r.ContentLength))
+	defer func() {
+		putReadTrackingBody(r.Body.(*readTrackingBody))
+	}()
```

Comment:
I think it's correct to check for the max value of ContentLength header. It prevents possible DDoS attacks.
```suggestion
        if int(r.ContentLength) < maxRequestBodySizeToRetry.IntN() {
           r.Body = getReadTrackingBody(r.Body, int(r.ContentLength))
	   defer func() {
		putReadTrackingBody(r.Body.(*readTrackingBody))
	}()
        }
```